### PR TITLE
Shortcut to special files

### PR DESCRIPTION
**Pull request by harshad1**
This PR adds the ability to create shortcuts to special files (Favourite etc on the home screen).

They are created as usual, like any other folder - long press and create shortcut

### CODE REVIEW

File: app/src/main/java/net/gsantner/markor/util/MarkorContextUtils.java [hunk 1]
```diff
@@ -46,13 +47,27 @@ public <T extends GsContextUtils> T applySpecialLaunchersVisibility(final Contex
         return thisp();
     }
 
+    private static int getIconResForFile(final File file) {
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/util/MarkorContextUtils.java:50]: suggestion: add @ nullable or nonnull, if it's nullable then handle if (null) in first condition

File: app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java [hunk 2]
```diff
@@ -37,6 +37,7 @@
 import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
 import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+import androidx.appcompat.widget.Toolbar;
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java:40]: unneeded import (automatic cleanup possible)

**Review by gsantner — COMMENTED** [review]
thanks

File: app/src/main/java/net/gsantner/markor/util/MarkorContextUtils.java (hunk 1 above)
Comment (harshad1) [reply app/src/main/java/net/gsantner/markor/util/MarkorContextUtils.java]: Fixed

File: app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java (hunk 2 above)
Comment (harshad1) [reply app/src/main/java/net/gsantner/opoc/frontend/filebrowser/GsFileBrowserFragment.java]: Fixed

**Review by gsantner — APPROVED** [review]
thanks, works!

**Comment by cinnabar777** [comment]
When using this feature the home screen shortcut opens a new window every time it's trapped instead of bringing up the already open window. 

I think it's due to having multiple windows option enabled in settings? 

Nonetheless the shortcut does open the file and in the last used state, but will open news window every time.

**Pull request merged**